validator/modules/onnx/__init__.py

```
@@ -0,0 +1,289 @@
+            input_data = input_data.astype(np.float32)
+
+            if len(input_data.shape) == 1:
+                input_data = input_data.reshape(1, -1)
+
+            outputs = self.model.run(None, {input_name: input_data})
+
+            return outputs[0].flatten()
+
+        except Exception as e:
+            print(f"Error during inference: {e}")
+            raise ValueError(f"ONNX inference failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)